app/services/save_practice_service.rb

```diff
@@ -190,16 +189,17 @@ def update_category_practices
           name = category[:name]
           parent_cat_id_param = category[:parent_category_id]
           # if the 'other' category has not yet been created, the :parent_category_id will be a string, so we need to find the corresponding parent category
-          parent_cat_id = parent_cat_id_param.to_i === 0 ? Category.where('lower(name) = ?', parent_cat_id_param.downcase).first.id : parent_cat_id_param
+          parent_cat_id = parent_cat_id_param.to_i === 0 ? Category.get_category_by_name(parent_cat_id_param).first.id : parent_cat_id_param
           unless name == ""
-            if destroy == 'false' && id.blank?
-              cate = Category.find_by(name: name.strip.downcase, is_other: true, parent_category_id: parent_cat_id)
-              cate = Category.create(name: name.strip.downcase, is_other: true, parent_category_id: parent_cat_id) unless cate.present?
+            if destroy != '1' && id.blank?
+              cate = Category.find_by(name: name.strip, is_other: true, parent_category_id: parent_cat_id)
```

Comment:
I think we should do something like:
``` 
cat_name = name.strip.downcase
cate = Category.where('lower(name) = ?', cat_name).where(is_other: true).first
```

Reply:
Cool - so ... this is the change right..
![image](https://user-images.githubusercontent.com/60527788/125518430-847771c1-c829-4134-abc3-711ec7b847b5.png)

Reply:
haha please ignore https://agilesix.slack.com/archives/C010LQ2V3MZ/p1626207673054800?thread_ts=1626201526.048100&cid=C010LQ2V3MZ 😅